From 99c61fc8beda697ae1d0ababa1cf6e357e539f40 Mon Sep 17 00:00:00 2001
From: Tian Zhang <tian.c.zhang@intel.com>
Date: Fri, 15 May 2015 16:15:09 +0800
Subject: [PATCH 078/403] Callback abstraction and refinement for vgtbuffer

Signed-off-by: Tian Zhang <tian.c.zhang@intel.com>

Add a function pointer 'insert_vmfb_entries' as a unified interface
for inserting VM framebuffer entries into dom0's PPGTT, so that to
hide hardware dependent implementation.

Also refined function 'gen6_ppgtt_insert_vmfb_entries' to get rid
of an unnecessary parameter.
---
 drivers/gpu/drm/i915/i915_gem_vgtbuffer.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_vgtbuffer.c b/drivers/gpu/drm/i915/i915_gem_vgtbuffer.c
index 02ef9a2..08fd859 100644
--- a/drivers/gpu/drm/i915/i915_gem_vgtbuffer.c
+++ b/drivers/gpu/drm/i915/i915_gem_vgtbuffer.c
@@ -199,14 +199,13 @@ i915_gem_vgtbuffer_ioctl(struct drm_device *dev, void *data,
 
 	i915_gem_object_init(obj, &i915_gem_vgtbuffer_ops);
 	obj->cache_level = I915_CACHE_L3_LLC;
+	obj->has_vmfb_mapping = true;
+	obj->pages = NULL;
 
 	struct i915_address_space *ggtt_vm = &dev_priv->gtt.base;
 	struct i915_vma *vma = i915_gem_obj_lookup_or_create_vma(obj, ggtt_vm);
 	vma->node.start = gtt_offset;
 
-	obj->has_vmfb_mapping = true;
-	obj->pages = NULL;
-
 	ret = drm_gem_handle_create(file, &obj->base, &handle);
 	/* drop reference from allocate - handle holds it now */
 	drm_gem_object_unreference(&obj->base);
-- 
1.7.10.4

